server/src/main/java/com/ncc/neon/services/ClusterService.java

```diff
@@ -23,16 +26,190 @@
+     *
+     * @param tabularQueryResult the original results
+     * @return the tabularqueryresult aggregated by numbers with the specifications given by the cluster clause
+     */
+    private TabularQueryResult aggregateNumber(TabularQueryResult tabularQueryResult) {
+        List<List<Object>> clusters = this.clusterClause.getClusters();
+
+        // establish keys
+        String fieldNameKey = this.clusterClause.getFieldNames().get(0);
+        String aggregationNameKey = this.clusterClause.getAggregationName();
+
+        List<Map<String, Object>> data = tabularQueryResult.getData();
+
+        // determine ordering
+        Map<String, Object> first = data.get(0);
+        Map<String, Object> last = data.get(data.size() - 1);
+        BigDecimal firstGroup = new BigDecimal(first.get(fieldNameKey).toString()); // assume number because 'number' fieldType
+        BigDecimal lastGroup = new BigDecimal(last.get(fieldNameKey).toString());
+        String order = "asc";
+        if (firstGroup.compareTo(lastGroup) == 1) {
+            order = "des";
+        }
+
+        // determine count
+        int count = 50; // default count
+        if (this.clusterClause.getCount() != 0) {
+            count = this.clusterClause.getCount();
+        }
+
+        // data is small enough
+        if (clusters == null && count > data.size()) {
+            return tabularQueryResult;
+        }
+
+        // find new clusters
+        List<Map<String, Object>> newData = getNewDataBins(fieldNameKey, firstGroup, lastGroup, count, clusters);
+
+        // take care of extra keys that may be present
+        Map<String, Object> extraKeySets = getExtraKeySetsMap(fieldNameKey, aggregationNameKey, data);
+
+        // aggregate data into the new clusters
+        aggregateNumbersInNewData(fieldNameKey, aggregationNameKey, data, newData, extraKeySets);
+
+        this.clusterClause = null;
+        return new TabularQueryResult(newData);
+    }
+
+    /**
+     * Calculates the spacing between the different bins in the new data set results.
+     *
+     * @param fieldNameKey the key that corresponds with the field name
+     * @param firstGroup the first group of data in the original result data
+     * @param lastGroup the last group of data in the original result data
+     * @param count how many bins there should be in the new data set
+     * @param clusters clusters specified by the cluster clause if available
+     * @return the new data set ready to be filled
+     */
+    private List<Map<String, Object>> getNewDataBins(String fieldNameKey, BigDecimal firstGroup, BigDecimal lastGroup,
+                                                     int count, List<List<Object>> clusters) {
+        List<Map<String, Object>> newData = new ArrayList<>();
+        if (clusters == null) {
+            BigDecimal gap = (lastGroup.subtract(firstGroup)).divide(new BigDecimal(count));
+            BigDecimal step = new BigDecimal(".0001");
```

Comment:
Could we make `".0001"` a global variable, so it's easier to change in the future, if needed?

Reply:
Fixed.